trie/leavesRetriever/dfsTrieIterator/dfsTrieIterator.go

```diff
@@ -77,70 +111,51 @@ func (it *dfsIterator) GetLeaves(numLeaves int, ctx context.Context) (map[string
 			return nil, err
 		}
 
-		childrenSize := uint64(0)
 		for _, childNode := range childrenNodes {
 			if childNode.IsLeaf() {
 				key, err := childNode.GetKeyBuilder().GetKey()
 				if err != nil {
 					return nil, err
 				}
 
-				retrievedLeaves[string(key)] = string(childNode.GetData())
+				keyValHolder, err := leavesParser.ParseLeaf(key, childNode.GetData(), childNode.GetVersion())
+				if err != nil {
+					return nil, err
+				}
+
+				hexKey := hex.EncodeToString(keyValHolder.Key())
```

Comment:
is this change treated on upper levels? since before the keys/values were string, not hex

Reply:
The endpoint `/address/keys` will return the key-value pairs as hex (`node/node.go` L360). I wanted this endpoint to return key-val pairs in the same way.